File: api/src/main/java/net/blueshell/api/repository/contribution/ContributionReminderRepository.java
package net.blueshell.api.repository.contribution;

import net.blueshell.api.base.BaseRepository;
import net.blueshell.api.model.contribution.ContributionPeriod;
import net.blueshell.api.model.contribution.ContributionReminder;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ContributionReminderRepository extends BaseRepository<ContributionReminder> {
    List<ContributionReminder> findByContributionPeriod(ContributionPeriod contributionPeriod);
}

File: api/src/main/kotlin/net/blueshell/api/repository/contribution/ContributionReminderRepository.kt
package net.blueshell.api.repository.contribution

import net.blueshell.api.model.contribution.ContributionReminder
import net.blueshell.api.repository.base.BaseRepository
import org.springframework.stereotype.Repository

@Repository
interface ContributionReminderRepository : BaseRepository<ContributionReminder, ContributionReminder.Id> {
    fun findById_ContributionPeriodId(contributionPeriodId: Long): MutableList<ContributionReminder>
}
